pkg/workload/admissionchecks.go

```diff
@@ -166,9 +164,8 @@ func HasAllChecks(wl *kueue.Workload, mustHaveChecks sets.Set[string]) bool {
 
 // HasRetryChecks returns true if any of the workloads checks is Retry
 func HasRetryChecks(wl *kueue.Workload) bool {
-	for i := range wl.Status.AdmissionChecks {
-		state := wl.Status.AdmissionChecks[i].State
-		if state == kueue.CheckStateRetry {
+	for _, ac := range wl.Status.AdmissionChecks {
```

Comment:
nit: I prefer to revert to avoid distractions

Reply:
Done